File: 백준/Bronze/27590. Sun and Moon/Sun and Moon.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
let [ds, ys] = input[0].split(" ").map(Number);
let [dm, ym] = input[1].split(" ").map(Number);
let sun = ys - ds;
let moon = ym - dm;
while(sun != moon){
    sun < moon ? sun += ys : moon += ym;
}
console.log(sun);

File: 백준/Bronze/27590. Sun and Moon/Sun and Moon.py
ds, ys = list(map(int, input().split()))
dm, ym = list(map(int, input().split()))
sun = ys - ds
moon = ym - dm
while sun != moon :
    if sun < moon :
        sun = sun + ys
    else :
        moon = moon + ym
print(sun)